reana_job_controller/job_manager.py

```diff
@@ -61,7 +63,8 @@ def wrapper(inst, *args, **kwargs):
             inst.before_execution()
             backend_job_id = fn(inst, *args, **kwargs)
             inst.create_job_in_db(backend_job_id)
-            inst.cache_job()
+            if CACHE_ENABLED:
+                inst.cache_job()
```

Comment:
Should we modify `cache_job()` implementation at the same time to simply raise NotImplemented or to return nothing, with a code comment that the current caching implementation is broken, so that nobody would be penalised in performance when tempted to switch it on?